mono/mini/method-to-ir.c

```diff
@@ -377,11 +377,20 @@ break_on_unverified (void)
 		G_BREAKPOINT ();
 }
 
+static void
+clear_error (MonoError *error)
+{
+	mono_error_cleanup (error);
+// Hijack mono_error_cleanup function to be local macro that also initializes for reuse.
```

Comment:
Just change the uses of `mono_error_cleanup` to `clear_error` instead of doing this hack.

Reply:
I agree and disagree. That is equivalent to what I have. Do we want to permanently take the position that within method-to-ir.c, all cleanup allows reuse, or leave it easily changed so we can go back and debug and find the reuses? I would accept several answers here:
1. Change main cleanup to init.
2. Change main cleanup to init, with intention of going back to debug.
3. Debug method-to-ir.c and find every reuse, before commit.
4. Commit method-to-ir.c in either form (central or non-central reinit), with intent or not to later debug it, after commit.
I can try to head this off by doing #3 -- debug in the shorter term.
I'd rather not sprinkle all the cleanups around if intent is to later debug, but that is a choice -- later debug would start by removing them all.

Reply:
For the sake of simplicity, just make it explicit that you are calling `clear_error` and please please please avoid all this magic. Who is going to read this code after you? Most likely not you, so please make it simpler for the next person and *simply* call `clear_error` instead of "hijacking mono_error_cleanup".